### ISSUE TITLE
[no-implied-eval] False positive with fn.bind()

### ISSUE THREAD

**Issue by scottohara** [typescript-eslint/typescript-eslint#2201]
This issue is very similar to #1642, except in this case it occurs when calling `.bind()` on a function.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-implied-eval": "error"
  }
}
```
The following code was directly copied from the "examples of correct code" in the [README](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-implied-eval.md):

```TS
const foo = {
  fn: function () {},
};
setTimeout(foo.fn.bind(this), 100);
```

**Expected Result**
No errors

**Actual Result**
```
error  Implied eval. Consider passing a function  @typescript-eslint/no-implied-eval
```

**Additional Info**
The type returned when calling `fn.bind()` seems to differ depending on whether Typescript's `strictBindCallApply` option is enabled or not.

When `strictBindCallApply: false`, calling `fn.bind()` produces an `any` type.
When `strictBindCallApply: true`, calling `fn.bind()` produces the same type as `fn`
(e.g. in the repro above, it would be `() => void`).

Regardless, in both cases the rule is triggered and warns of implied eval.

Either this is truly another false positive (like #1642), or the README should be updated to remove this particular example (if calling `.bind()` is legitimately considered a type of implied eval?).

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.1.0` |
| `@typescript-eslint/parser`        | `3.1.0` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `14.2.0` |
| `npm`                              | `6.14.5` |

**Comment by scottohara** [comment]
Actually no...I'm mistaken.

When `strictBindCallApply: true`, the rule does not trigger.

If this is as intended, this issue can be closed.
